shenyu-admin/src/main/java/org/apache/shenyu/admin/controller/ApiController.java

```diff
@@ -0,0 +1,141 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.admin.controller;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.shenyu.admin.mapper.ApiMapper;
+import org.apache.shenyu.admin.model.dto.ApiDTO;
+import org.apache.shenyu.admin.model.page.CommonPager;
+import org.apache.shenyu.admin.model.page.PageParameter;
+import org.apache.shenyu.admin.model.query.ApiQuery;
+import org.apache.shenyu.admin.model.result.ShenyuAdminResult;
+import org.apache.shenyu.admin.model.vo.ApiVO;
+import org.apache.shenyu.admin.service.ApiService;
+import org.apache.shenyu.admin.utils.ShenyuResultMessage;
+import org.apache.shenyu.admin.validation.annotation.Existed;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * this is api controller.
+ */
+@Validated
+@RestController
+@RequestMapping("/api")
+public class ApiController {
+
+    private final ApiService apiService;
+
+    public ApiController(final ApiService apiService) {
+        this.apiService = apiService;
+    }
+
+    /**
+     * query apis.
+     *
+     * @param apiPath     api path.
+     * @param state       state.
+     * @param tagId       tagId.
+     * @param currentPage current page.
+     * @param pageSize    page size.
+     * @return {@linkplain ShenyuAdminResult}
+     */
+    @GetMapping("")
+    public ShenyuAdminResult queryApis(final String apiPath, final Integer state,
+                                       final String tagId,
+                                       @NotNull final Integer currentPage,
+                                       @NotNull final Integer pageSize) {
+        CommonPager<ApiVO> commonPager = apiService.listByPage(new ApiQuery(apiPath, state, tagId, new PageParameter(currentPage, pageSize)));
+        return ShenyuAdminResult.success(ShenyuResultMessage.QUERY_SUCCESS, commonPager);
+    }
+
+    /**
+     * detail plugin.
+     *
+     * @param id plugin id.
+     * @return {@linkplain ShenyuAdminResult}
+     */
+    @GetMapping("/{id}")
+    @RequiresPermissions("system:api:edit")
+    public ShenyuAdminResult detailApi(@PathVariable("id")
```

Comment:
Why do query operations require permission?

Reply:
fixed it